File: 2997.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        count = 0
        for bit in range(20):
            xorVal = 0
            for i in nums:
                xorVal ^= ((1<<bit) & i)
            kbitVal = (1<<bit) & k
            if kbitVal != xorVal:
                count+=1


        return count

File: 2997.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int count = 0;
        for (int bit = 0; bit <= 20; bit++){
            int xorVal = 0;
            for (auto i: nums){
                int bitVal = (1<<bit)&i;
                xorVal = xorVal ^ bitVal;
            }
            int kxorVal = ((1<<bit)&k);
            if (xorVal != kxorVal){
                count++;
            } 
        }
        return count;
    }
};